### PR TITLE
[Mbed] Fix device commisioning - BLE adv and LWIP PKTINFO

### PR DESCRIPTION
**Pull request by ATmobica**
#### Problem
We need to fix some issues to improve the device commissioning process according to a new specification.

1. Enable LWIP recvinfo feature - is necessary after Mbed-os update #13416 
2. Add start BLE advertising after push button
3. Apps Cmake files refactor - use chip_configure_data_model to include generated files

#### Change overview
Add LWIP recvinfo flag to mbed application
Add chip_configure_data_model to lock-app and lighting-app cmake
Add start BLE advertising after push button
Update mbed documentation

#### Testing
Build and run Mbed examples, commissioning device with chip-device-controller

### CODE REVIEW

**Review by pan- — APPROVED** [review]

**Review by Damian-Nordic — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
fast track: platform specific change.

**Review by bzbarsky-apple — APPROVED** [review]

**Comment by ATmobica** [comment]
@andy31415 @bzbarsky-apple Can we merge this PR?

**Review by pan- — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@ATmobica I can't merge it because Github thinks it has conflicts?

**Comment by ATmobica** [comment]
@bzbarsky-apple Conflicts resolve

**Comment by ATmobica** [comment]
@andy31415 @bzbarsky-apple @woody-apple 
Can we finally merge this PR? I had to rebase and resolve conflicts due to the changes in the clusters many times. With these changes no more CmakeList files changes will be required.

**Comment by bzbarsky-apple** [comment]
Needs to be rebased on top of #14384 once that lands to fix CI.  :(

**Comment by ATmobica** [comment]
> Needs to be rebased on top of #14384 once that lands to fix CI. :(
> 
> I'm really sorry about the pain here....

@bzbarsky-apple Ok, I'm waiting to merge this fix.

**Pull request merged**
